sql/item_cmpfunc.h

```diff
@@ -1003,6 +1003,20 @@ class Item_func_opt_neg :public Item_bool_func
 
 class Item_func_between :public Item_func_opt_neg
 {
+  /*
+    If the types of the arguments to BETWEEN permit, then:
+
+    WHERE const1 BETWEEN expr2 AND field1
+      can be optimized to:
+    WHERE const1 <= field1
+
```

Comment:
Can you please also add a comment the other way around:
`WHERE const1 BETWEEN field1 AND expr2`
can be optimized like if it was just:
`WHERE const1 >= field1`
?

Reply:
Yes, it supersedes. Thanks.